src/main/java/com/alvarium/annotators/vulnerability/OsvApiHandler.java

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright 2023 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package com.alvarium.annotators.vulnerability;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+
+import com.google.gson.Gson;
+
+public class OsvApiHandler implements VulnerabilityApiHandler {
+    final private Gson gson = new Gson();
+    final private String endpoint;
+
+    public OsvApiHandler() {
+        this.endpoint = "https://api.osv.dev/v1/querybatch";
+    }
+
+    private OsvBatchResponse makeApiCall(Map<String,String> packages) throws VulnerabilityException {
+        try {
+            // Create the HTTP client and POST request
+            CloseableHttpClient httpClient = HttpClients.createDefault();
+            HttpPost httpPost = new HttpPost(this.endpoint);
+
+            // Set the request body
+            OsvBatchRequest req = new OsvBatchRequest(packages);
+            StringEntity entity = new StringEntity(gson.toJson(req), StandardCharsets.UTF_8);
+            httpPost.setEntity(entity);
+
+            // Set the content type header
+            httpPost.setHeader("Content-Type", "application/json");
+
+            // Execute the request
+            CloseableHttpResponse response = httpClient.execute(httpPost);
+

```

Comment:
In your testing, what kind of response times have you observed when making this call? If the call were to fail, would that halt the build -- or would the build proceed with a corresponding confidence penalty?

Reply:
1. While testing myself on my PC, I got responses after 600ms on average. The API docs have some info on their targets for latency and availability: https://google.github.io/osv.dev/faq/#what-are-osvs-service-level-objectives-slos

2. @o-eissa would be better to answer here since he worked on this stuff but I can't seem to @tag him here. Anyway, looking at the code it seems that the annotator would not be published at all if the api call fails, so no confidence penalty would be applied. If

Reply:
Yes as Ali stated there's no confidence penalty in that case, if the call failed the annotator wouldn't continue execution and the annotation wouldn't be published.

Reply:
I think ideally the annotator would always fire if it's configured to do so. If the API call were to fail due to an error, then the error should be logged and the annotator would return `false` which would then decrease the confidence value.

The problem with that is that the Java SDK doesn't return multiple values from its functions like the Go SDK does. On the other hand, we could pass a `logger` instance into the constructors of the annotators. It occurs to me that this is probably a ubiquitous issue for all annotators in the Java SDK. If any annotator were to throw an exception, the exception is logged and it's like the annotator never fired -- when it should probably be counted as a "false". What do you guys think?

If additional code changes are required, we should create an issue and put them in another PR.

Reply:
I agree... I think from an end-user perspective, I should have the same confidence in a piece of data that was not able to verify a precondition as another piece of data that did not pass that precondition. 

For example, if I enable PKI annotators and send out 2 different pieces of data:
- Data A: Uses signing key that has been tampered with, the annotation is not satisfied and confidence should drop
- Data B: Payload was sent without a signature, the annotation will throw an exception. Confidence should definitely drop here as well

In the light of this, I opened #111.